src/librustc_mir/interpret/visitor.rs

```diff
@@ -221,7 +221,7 @@ macro_rules! make_value_visitor {
                         let inner = v.project_downcast(self.ecx(), idx)?;
                         trace!("walk_value: variant layout: {:#?}", inner.layout());
                         // recurse with the inner type
-                        return self.visit_field(v, idx, inner);
+                        return self.visit_field(v, 0, inner);
```

Comment:
cc @RalfJung I think this move to type-safe indices already found a bug :smiley:

Reply:
maybe this should even just be `visit_value`

Reply:
Not a bug, but a hack. :)  This was intentional.  `visit_field` is used by the validating visitor to keep track of the path, so that we can print a nice error. We do want enum variants included in there.

Reply:
I added a `visit_variant` method to the visitor

Reply:
I agree that should work. I do not like the code duplication between validation's `visit_field` and `visit_variant` though, can you find a way to avoid that?